assert/assertions_test.go

```diff
@@ -2786,6 +2786,17 @@ func TestEventuallyWithTTrue(t *testing.T) {
 	Len(t, mockT.errors, 0)
 }
 
+func TestEventuallyWithT_ConcurrencySafe(t *testing.T) {
```

Comment:
It may be beneficial to have an additional test which explicitly exposes the bug this PR fixes. [Here](https://github.com/stretchr/testify/pull/1438/files#diff-7190065248695df477329c4cada2d71f323ab7940ecffff2d3d5fa43caf79427R2766) is an example -- if the first check of the condition is allowed to complete quickly and the next takes longer than the timeout, it is very easy to expose the bug.

Reply:
Thank you for the suggestion. Yes, this reproduces the issue of reporting no errors due to the `Reset()` call just before running another condition. Allowed myself to copy the test and supplement that with comments. 👍